components/eamxx/src/physics/mam/mam_coupling.hpp

```diff
@@ -549,7 +550,8 @@ mam4::Prognostics interstitial_aerosols_for_column(const AerosolState& dry_aero,
     progs.n_mode_i[m] = ekat::subview(dry_aero.int_aero_nmr[m], column_index);
     for (int a = 0; a < num_aero_species(); ++a) {
       if (dry_aero.int_aero_mmr[m][a].data()) {
-        progs.q_aero_i[m][a] = ekat::subview(dry_aero.int_aero_mmr[m][a], column_index);
+        int prg_idx = mam4::utils::get_prognostics_index(m,a);//Index of HAERO's prognostics 
```

Comment:
I believe, we do not need `mam4::utils::get_prognostics_index(m,a)`. The species ordering in each mode  is equal in  mam4xx and emaxx.

Reply:
In addition, see my comment in[ PR #308](https://github.com/eagles-project/mam4xx/pull/308) in mam4xx. 
> I reviewed this part of the code and found that we cannot use ` mam4::mode_aero_species(mode, species_index)` to obtain the index of the species in each mode. ` mam4::mode_aero_species(mode, species_index)` returns the aero_id, which is an enum from [here](https://github.com/eagles-project/mam4xx/blob/7d14b8a71166c48327f1839c46c0f974134a631f/src/mam4xx/aero_modes.hpp#L130).
> 
> Because mam4xx prognostics structure has same ordering of species per node that state_q and qqcw, we do not need to add this function; however, we could use the function that @jeff-cohere have implemented [here](https://github.com/eagles-project/mam4xx/blob/7d14b8a71166c48327f1839c46c0f974134a631f/src/mam4xx/aero_modes.hpp#L249) and it is used [here](https://github.com/eagles-project/mam4xx/blob/7d14b8a71166c48327f1839c46c0f974134a631f/src/mam4xx/utils.hpp#L163).

Reply:
You are right. I have removed it.